doc/src/ch02-01-generic.md

```diff
@@ -1 +1,23 @@
 # Monitoring a file descriptor
+The `Generic` event source wraps a file descriptor ("fd") and fires its callback any time there are events on it ie. becoming readable or writable, or encountering an error. It's pretty simple, but it's what every other event source is based around. And since the platforms that calloop runs on expose many different kinds of events via fds, it's usually the key to using those events in calloop.
+
+For example on Linux, fd-based interfaces are available for GPIO, I2C, USB, UART, network interfaces, timers and many other systems. Integrating these into calloop starts with obtaining the appropriate fd, creating a `Generic` event source from it, and building up a more useful, abstracted event source around that. A detailed example of this is [given for ZeroMQ](ch04-00-a-full-example-zeromq.md). 
+
+## Creating a Generic source
+
+Creating a `Generic` event source requires three things:
+- The fd itself, of course
+- The ["interest"](api/calloop/struct.Interest.html) you have in this descriptor — this means, whether you want to generate events when the fd becomes readable, writeable, or both
+- The ["mode"](api/calloop/enum.Mode.html) of event generation - level triggered or edge triggered. If you are familiar with hard descriptor languages, you can think of this as the sensitivity.
+
+The easiest constructor to use is the [`new()`](api/calloop/generic/struct.Generic.html#method.new) method, but if you need control over the [associated error type](ch02-06-errors.md) there is also [`new_with_error()`](api/calloop/generic/struct.Generic.html#method.new_with_error).
+
+## Ownership
+
+It's important to remember that file descriptors have no concept of ownership attached to them in Rust — they are simply bare integers. Typically (eg. in the standard library) they would be an underlying implementation detail of another type that *did* encode ownership somehow, but in order to integrate so closely with the systems they expose, we must use them (and their platform APIs) directly.
+
+This means you need to do at least two things:
+- follow the rules of the API you obtain the fd from
+- wrap them in a type that manages ownership appropriately
+
+For example, the `Ping` event source in calloop uses a reference-counted `CloseOnDrop` wrapper to ensure that the underlying fd is closed when all clones of the ping's generic event source are, themselves, dropped.
```

Comment:
I'm not a huge fan of talking about `calloop`'s internals in the documentation like this. Let's maybe instead encourage using appropriate types, like for example illustrate how `Generic<Socket>` will close the socket on drop?

And generally, `Generic` is not only about fds, but about types which are backed by fds.

Reply:
> like for example illustrate how Generic<Socket> will close the socket on drop?

Ah that's a good point. I will redo this part.

Reply:
I redid this part with more emphasis on using wrapper types, with an example from the standard library. I didn't mention the `OwnedFd` and `BorrowedFd` types because they're still unstable, but when they stabilise they might warrant a mention here too.